pkg/provider/azure_managedDiskController.go

```diff
@@ -251,35 +252,33 @@ func (c *ManagedDiskController) CreateManagedDisk(options *ManagedDiskOptions) (
 }
 
 //DeleteManagedDisk : delete managed disk
-func (c *ManagedDiskController) DeleteManagedDisk(diskURI string) error {
-	diskName := path.Base(diskURI)
+func (c *ManagedDiskController) DeleteManagedDisk(ctx context.Context, diskURI string) error {
 	resourceGroup, err := getResourceGroupFromDiskURI(diskURI)
 	if err != nil {
 		return err
 	}
 
-	ctx, cancel := getContextWithCancel()
-	defer cancel()
-
 	if _, ok := c.common.diskStateMap.Load(strings.ToLower(diskURI)); ok {
 		return fmt.Errorf("failed to delete disk(%s) since it's in attaching or detaching state", diskURI)
 	}
 
+	diskName := path.Base(diskURI)
 	disk, rerr := c.common.cloud.DisksClient.Get(ctx, resourceGroup, diskName)
 	if rerr != nil {
 		if rerr.HTTPStatusCode == http.StatusNotFound {
 			klog.V(2).Infof("azureDisk - disk(%s) is already deleted", diskURI)
 			return nil
 		}
-		return rerr.Error()
+		// ignore GetDisk throttling
```

Comment:
So what if it throttles when deleting? Why should we ignore this?

Reply:
original error case is rate limited by disk driver, we should ignore and continue delete disk.
Anyway, if read operation is throttled or rate limited, should not impact write operation(delete disk), and read operation is quite more frequent than write operation, we should expect such case and continue delete disk.

Reply:
if deleting is throttled, then return error since delete failed.

GetDisk is quite more frequent than delete disk operation, we should expect such case, and continue delete disk if GetDisk is throttled. This function is delete disk, we could ignore GetDisk if it's throttled.